utils.ts

```diff
@@ -42,33 +42,17 @@ export function check_encoded_allocation(encoded_allocation: u32): ErrorCode {
 
 // writes string to memory, then returns encoded allocation ref
 export function serialize(val: string): u32 {
-  let dataLength = val.length;
-  // each char takes two bytes, encoded
-  let ptr = memory.allocate(dataLength << 1);
-  //checkMem();
-  for (let i = 0; i < dataLength; ++i) {
-    store<u16>(ptr + i, val.charCodeAt(i));
-  }
-  let encoded_allocation = u32_merge_bits(ptr as u16, dataLength as u16);
-  return encoded_allocation;
+  let ptr = val.toUTF8();
+  let length = val.length;
```

Comment:
Are you need length in UTF16 or UTF8? In last case you should use `val.lengthUTF8` getter instead `val.length`

Reply:
Good save. This would have been disastrous for multi-byte characters

Reply:
:pray

Reply:
actually, just based on testing, it appears that the way you have it works @willemolding, and the way it's suggested doesn't